### ISSUE TITLE
Playing Videos protected with Nginx "X-Accel-Redirect"

### ISSUE THREAD

**Issue by 06GitHub** [Selz/plyr#453]
Thanks for plyr.js player which is an awesome player.

Question : I am using nginx as web server and need to protect videos from viewing/download with ‘X-Accel-Redirect’ feature https://www.nginx.com/resources/wiki/start/topics/examples/x-accel/

Users can access videos only if they have some rights (eg have subscribed to training session).

Nginx server configuration relies on ‘internal’ keyword :
    _location /media_protected/videos/ {
        internal;
        alias /home/myproject/media_protected/videos/
    }_

and response http header must contain ‘X-Accel-Redirect’ with video url, eg :
_HttpResponse['X-Accel-Redirect'] = ‘/media_protected/videos/myvideo.mp4’_

**How can HttpResponse header ['X-Accel-Redirect'] be set by plyr.js player ?**

Thanks.

**Comment by SamPotts** [comment]
HTTP headers can't be set unless the video is loaded via XHR/AJAX which would complicate things. Maybe something to consider later but I'd find another way to do it if you can.

**Comment by 06GitHub** [comment]
Thanks for feedback. Indeed I have tried to set up HTTP headers in Javascript by updating some parts of plyr.js code but no success.

So far I have implemented a workaround : videos files names and audio file names are generated with random characters (eg "myvideo_Ab$-4n&fl.mp4"). This makes them more difficult to guess but is not as secure as native Nginx "X-Accel-Redirect" protection (similar to Apache "X-Sendfile" one), unfortunately.

Congratulations again for your plyr.js : player works well on all platforms I use (mobile phones, laptops, desktops, tablets, Windows/IOS/Android, …), is very user friendly and nice looking. Good job :-)
